### DISCUSSION TITLE
Vague Error Opening Hubs World BUT with Console errors attached

### DISCUSSION THREAD

**Discussion by mishawagon** [mozilla/hubs#3197] [General Questions & Discussion] [answered]
Dear Community,

Amazing work on the Hubs/Spoke platform: it's revolutionary! 
We are publishing a Spoke world no probs, but when it comes to visit it in Hubs there is a vague error, however there are some JS errors in console I don't know how to parse. Is it a specific model that's giving us heck? When publishing, the summary says polygons and materials are Moderate, textures are low as in the screenshot : https://drive.google.com/file/d/1nGBJNuy61fXQFQY00iGLuNzWxKmhM0C_/view?usp=sharing

The actual world:
https://hubs.mozilla.com/spoke/projects/AKHcX5b

The console errors: https://drive.google.com/file/d/1ff0SqjtBMzRyLut5KrimLI4ifcTbNaqg/view?usp=sharing

Any additional info on how we can debug this would be super helpful!

**Comment by robinkwilson** [comment]
This might be a bug if it relates to the Content Security Policy and one of your models hosted is running into issues. 

To pinpoint the source: 
1. Can you look at the collapsed Content Security Policy error and see what urls the console is complaining about?
1. Next, look at your Network tab, are there any failed requests?
1. If you're using a Sketchfab model (or other services), have you added your API keys to the HC Admin Panel > Server Settings > API keys menu? 

Thank you!

**Comment by mishawagon** [comment]
@robin-k-wilson thank you very much. I'm checking the Content Security Policy errors and see these requests https://:443 and wss://:443 being potentially problemantic, but other than that I'm not sure what to look for. Network tab doesn't show any errors. Ok so I don't understand the HC Admin Panel, do I need to rent AWS and get Hubs Cloud for this? The Hubs Space: [](https://hubs.mozilla.com/Tyo7K5Q/gamer-church-test)
Thanks again for your time!

The source list for Content Security Policy directive 'connect-src' contains an invalid source: '<URL>'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
serialize-element.js:140 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
Wl @ serialize-element.js:140
serialize-element.js:140 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
Wl @ serialize-element.js:140
serialize-element.js:140 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
Wl @ serialize-element.js:140
serialize-element.js:140 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
Wl @ serialize-element.js:140
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
.*\.json$ namespace object:25 Uncaught (in promise) Error: Cannot find module './undefined.json'
    at .*\.json$ namespace object:25
.*\.json$ namespace object:25 Uncaught (in promise) Error: Cannot find module './undefined.json'
    at .*\.json$ namespace object:25
8The source list for Content Security Policy directive 'connect-src' contains an invalid source: '<URL>'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'wss://:443'. It will be ignored.
gamer-church-test:1 The source list for Content Security Policy directive 'connect-src' contains an invalid source: 'https://:443'. It will be ignored.

**Reply by robinkwilson** [reply]
Can you go to https://hubs.mozilla.com and create a room? I want to double check that you can connect to and enter a room.
Can you try making a new Spoke scene and import the most complex model you expect is having issues without any other assets to test that specific model?

No need to rent Hubs Cloud on aws -- just needed to confirm that you're on hubs.mozilla.com!

**Comment by mishawagon** [comment]
Yes, thank you I can and I see the Halloween space.

Great idea re: Spoke testing, will do!

Thank you R!

**Reply by robinkwilson** [reply]
:D Marking as answer until more context is given or the problem still persists. Click "Unmark as answer" when there's more info or you need to follow up.

**Comment by mishawagon** [comment, marked as answer]
Dear @robin-k-wilson we followed your advice and it worked! This is the offending model. We're kicking this ducky out of our Spoke world and sending it back south for the winter!
https://poly.google.com/view/bxU6M3D_B_x
Is it the CC-BY license I wonder..
So in the end your answer is the solution.

**Reply by mishawagon** [reply]
Not sure which one to mark as the answer but if you know please mark it. Thank you very much!

**Reply by robinkwilson** [reply]
Bye bye ducky!

**Reply by robinkwilson** [reply]
From answer above:

> Can you go to https://hubs.mozilla.com and create a room? I want to double check that you can connect to and enter a room.
> Can you try making a new Spoke scene and import the most complex model you expect is having issues without any other assets to test that specific model?
> 
> No need to rent Hubs Cloud on aws -- just needed to confirm that you're on hubs.mozilla.com!
